src/Volume.ts

```diff
@@ -1,96 +1,73 @@
-  /** Physical y size of a single *original* (not downsampled) pixel */
-  pixel_size_y: number;
-  /** Physical z size of a single pixel */
-  pixel_size_z: number;
-  /** Symbol of physical unit used by `pixel_size_(x|y|z)` fields */
-  pixel_size_unit: string;
-
-  // backwards compatibility: these are optional
-  /** Full x size of the volume. `tile_width` will be used if unspecified */
-  vol_size_x?: number;
-  /** Full y size of the volume. `tile_height` will be used if unspecified */
-  vol_size_y?: number;
-  /** Full z size of the volume. `tiles` will be used if unspecified */
-  vol_size_z?: number;
-  /** x offset of the volume subset into the volume */
-  offset_x?: number;
-  /** y offset of the volume subset into the volume */
-  offset_y?: number;
-  /** z offset of the volume subset into the volume */
-  offset_z?: number;
-
-  channels: number;
-  channel_names: string[];
-  channel_colors?: [number, number, number][];
+  version?: string;
+
+  /** XY size of the *original* (not downsampled) volume, in pixels */
+  // If we ever allow downsampling in z, replace with Vector3
+  originalSize: Vector2;
+  /** Number of rows and columns of z-slice tiles (not pixels) in the texture atlas */
+  atlasDims: Vector2;
+  /** Size of the volume, in pixels */
+  volumeSize: Vector3;
+  /** Size of the currently loaded subregion, in pixels */
+  regionSize: Vector3;
+  /** Offset of the loaded subregion into the total volume, in pixels */
+  regionOffset: Vector3;
+  /** Size of a single *original* (not downsampled) pixel, in spatial units */
+  pixelSize: Vector3;
+  /** Symbol of physical spatial unit used by `pixelSize` */
+  spatialUnit: string;
+
+  numChannels: number;
+  channelNames: string[];
+  channelColors?: [number, number, number][];
 
   times: number;
-  time_scale: number;
-  time_unit: string;
+  timeScale: number;
+  timeUnit: string;
 
   transform: {
-    translation: [number, number, number];
-    rotation: [number, number, number];
+    translation: Vector3;
+    rotation: Vector3;
   };
-  userData?: Record<string, unknown>;
-}
 
-export const getDefaultImageInfo = (): ImageInfo => {
-  return {
-    name: "",
-    version: "",
-    width: 1,
-    height: 1,
-    channels: 0,
-    tiles: 1,
-    pixel_size_x: 1,
-    pixel_size_y: 1,
-    pixel_size_z: 1,
-    pixel_size_unit: "",
-    channel_names: [],
-    channel_colors: [],
-    rows: 1,
-    cols: 1,
-    tile_width: 1,
-    tile_height: 1,
-    transform: {
-      translation: [0, 0, 0],
-      rotation: [0, 0, 0],
-    },
-    times: 1,
-    time_scale: 1,
-    time_unit: "",
-  };
+  userData?: Record<string, unknown>;
 };
 
-/* eslint-enable @typescript-eslint/naming-convention */
+export const getDefaultImageInfo = (): ImageInfo => ({
+  name: "",
+  version: "",
+  originalSize: new Vector2(1, 1),
+  atlasDims: new Vector2(1, 1),
+  volumeSize: new Vector3(1, 1, 1),
+  regionSize: new Vector3(1, 1, 1),
+  regionOffset: new Vector3(0, 0, 0),
```

Comment:
Nit: I wonder if references to `region` would be more descriptive as `subregion` instead, since it implies that it's a subset of the larger volume?